<launch>
  <!-- <include file="$(find darknet_ros)/launch/darknet_ros.launch">
   <arg name="image" value="/usb_cam/image_raw" />
  </include> -->

    <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>
  <arg name="image" default="/usb_cam/image_raw" />

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file"             default="$(find darknet_ros)/config/ros.yaml"/>
  <arg name="network_param_file"         default="$(find darknet_ros)/config/yolov2.yaml"/>
  <arg name="enable_opencv"         default="false"/>
  <arg name="enable_console_output" default="false"/>
  <!-- Load parameters -->
  <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <param name="image_view/enable_opencv"           value="$(arg enable_opencv)" />
    <param name="image_view/enable_console_output"           value="$(arg enable_console_output)" />
    
    <remap from="camera/rgb/image_raw"  to="$(arg image)" />
  </node>


  <include file="$(find usb_cam)/launch/usb_cam-test.launch"> 
  </include>
  
<node name="detection_publisher" pkg="darknet_ros_nodes" type="detection_publisher" output="screen" launch-prefix=""/> 
</launch>
